src/app/shares/viewshareaccount/viewshareaccount.component.ts

```diff
@@ -0,0 +1,354 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+
+/** Routing Imports */
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Imports */
+import { SharesService } from '../shares.service';
+
+/** Custom Models */
+import { PurchasedShare } from './purchased-shares.model';
+import { Charge } from './charges.model';
+import { Dividend } from './dividends.model';
+
+/** Common model for similar tables */
+export interface CommonTable {
+  details: string;
+  value: string;
+}
+
+/**
+ * View Share Account Component
+ */
+@Component({
+  selector: 'mifosx-viewshareaccount',
+  templateUrl: './viewshareaccount.component.html',
+  styleUrls: ['./viewshareaccount.component.scss']
+})
+export class ViewshareaccountComponent implements OnInit {
+
+  shareAccountId: any;
+  shareAccountDetails: any;
+  date: any = {};
+  staffData: any = {};
+  status: any;
+  choice: any;
+  dividends: any;
+  showDividends: any;
+  dividendsTableDataSource: Dividend[] = [];
+  chargeAction: any;
+  chargePayAction: any;
+  sharesPendingForApproval: any;
+  purchasedSharesTableShow: any;
+  purchasedSharesTableDataSource: PurchasedShare[] = [];
+  purchasedShares: any;
+  charges: any;
+  chargeTableShow: any;
```

Comment:
```suggestion
  chargeTable: boolean;
```
Maintain consistency with the variable names. Update in all cases.
Can't the respective data source variables be used in itself within ng if to ensure visibility? In that case, you should remove these variables completely and make use of that. Multiple unnecessary variables should not be created.